x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx

```diff
@@ -239,6 +240,14 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
   const [rule, setRule] = useState<Rule | null>(null);
   const isLoading = ruleLoading && rule == null;
 
+  const [isStartingJobs, setIsStartingJobs] = useState(false);
+  const { startMlJobs } = useStartMlJobs();
+  const startMlJobsIfNeeded = useCallback(async () => {
+    setIsStartingJobs(true);
+    await startMlJobs(rule?.machine_learning_job_id);
+    setIsStartingJobs(false);
+  }, [rule, startMlJobs]);
```

Comment:
Can we get rid of the code duplication by moving the following into the `useStartMlJobs` hook:

- `const [isStartingJobs, setIsStartingJobs] = useState(false);`
- error handling: in case of an exception
  - making sure `isStartingJobs` is reset to false 
  - we show a callout
- (if needed) `startMlJobs` should return a result of its execution

Reply:
Yup, we can do that. Though, the exception handling happens inside the `enableDatafeed`, so we don't need to do another try/catch inside `useStartMlJobs`

Reply:
done